core/src/main/java/com/datastax/oss/driver/api/core/type/DataTypes.java

```diff
@@ -51,13 +53,24 @@ public class DataTypes {
   public static final DataType TINYINT = new PrimitiveType(ProtocolConstants.DataType.TINYINT);
   public static final DataType DURATION = new PrimitiveType(ProtocolConstants.DataType.DURATION);
 
+  /* Vector types are currently expressed as a custom type but the class name in that custom type
+   * if followed by the number of dimensions for this specific type of vector enclosed by parentheses.
+   * A vector with M dimensions is of a different type than a vector with N dimensions (M != N) so
+   * we include the number of dimensions as part of the type.  */
+  private static final Pattern VECTOR_PATTERN =
+      Pattern.compile(CqlVectorType.CQLVECTOR_CLASS_NAME + "(\\d+)");
```

Comment:
I don't _love_ the idea of a regex compare here but it's not as bad as it appears at first blush.  We're only doing this for custom types so hopefully it should be a pretty rare event even for users of the vector type.  More importantly, most of the regex in question is a long literal string so eval'ing it as a regex shouldn't be vastly less performant than a base string compare.  Since it gives us the ability to extract the dimensions easily it seemed like a reasonable compromise, although we certainly can implement the extraction logic manually if need be.

Reply:
The regex is no longer used; we've addressed this issue via basic string ops.